src/sqlfluff/dialects/dialect_spark3.py

```diff
@@ -783,3 +783,117 @@ class JoinClauseSegment(BaseSegment):
     get_eventual_alias = ansi_dialect.get_segment(
         "JoinClauseSegment"
     ).get_eventual_alias
+
+
+@spark3_dialect.segment(replace=True)
+class TableExpressionSegment(BaseSegment):
+    """The main table expression e.g. within a FROM clause."""
+
+    type = "table_expression"
+    match_grammar = OneOf(
+        Ref("ValuesClauseSegment"),
```

Comment:
This had to come before `FunctionSegment`, or else it would try to interpret `values (1,2),(3,4)` as a function call

Reply:
Makes sense, that's the right approach 👍